core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -223,9 +225,13 @@ public void testToQuery() throws IOException {
 
         Query expectedQuery = createExpectedQuery(testQuery, context);
         Query actualQuery = testQuery.toQuery(context);
-        assertThat(actualQuery, instanceOf(expectedQuery.getClass()));
-        assertThat(actualQuery, equalTo(expectedQuery));
-        assertLuceneQuery(testQuery, actualQuery, context);
+        if (expectedQuery == null) {
```

Comment:
In the original FilteredQueryParser.parse(...) if both nested queries were parsed to null we returned null. This will change with #11915